lib/rules/no-is-mounted.js

```diff
@@ -41,7 +41,7 @@ module.exports = {
         }
         if (
           callee.object.type !== 'ThisExpression'
-          && (!('name' in callee.property) || callee.property.name !== 'isMounted')
+          || (!('name' in callee.property) || callee.property.name !== 'isMounted')
```

Comment:
```suggestion
          || !('name' in callee.property)
          || callee.property.name !== 'isMounted'
```

No need to wrap it in parentheses anymore (Also posted in https://github.com/jsx-eslint/eslint-plugin-react/pull/3731/files#r1756585084)

Reply:
Good point, removing the parentheses makes the check a bit more readable. I've updated the PR.